SVG back ground

VissitingLiam opened the pull request:
Added SVG background. 

Had CNC machine animations drawing the line of the CPU.

Still things to do of course, just want to hear what you have to say.

VissitingLiam committed:
SVG back ground

Adder svg back ground

VissitingLiam commented:
Sorting out dynamic loading and positioning

VissitingLiam committed:
SVG alingment

Fixed alignment of svg and logo

Vaperizer2 commented on src/components/custom/CncSvg.tsx:
please look into a minifyer for whatever that import is. there has to be a better way

VissitingLiam replied:
Will look into it.

Vaperizer2 commented on src/App.tsx:
```diff
@@ -1,149 +1,196 @@
 
+
 function App() {
+  const heroRef = useRef<HTMLDivElement | null>(null);        // hero wrapper (for scroll)
```
what does any of this used for, i dont see any reactivity on scroll or mouse movemnet/

VissitingLiam replied:
was planning on doin something with it but, just currently trying to fix other issues will remove them for now

Vaperizer2 replied:
ah ok, just cloned the code to see if i missed the use but if its still under dev don't stress

Vaperizer2 commented on src/App.tsx:
- [x] Remove old SVG background
- [x] Add a gloss behind the hero text to not let the cpu svg block readability.
- [ ] Please look at the centering of some elements, its off in a few places
- [ ] Remove animation on Logo on CPU, just looks weird. (If you really want it, animate on scroll using tailwind)
- [ ] please let the cnc routers disappear when done or run off screen, looks bad when they just sit there.
- [ ] Maybe have the start staggered a bit so its not so visually crowded.
- [x] Services component needs to match the same background as the rest

consider moving the hero to its own component and extract the needed css to a component specific file

Vaperizer2 commented on src/App.tsx:
```diff
@@ -1,149 +1,196 @@
+      if (rafRef.current == null) rafRef.current = requestAnimationFrame(doUpdate);
+    };
+
+    wrapper.addEventListener('mousemove', onMouseMove, { passive: true });
```
there is a better way to handle event listening. See AbortControllers in React Doc

Vaperizer2 reviewed (changes requested):
Looks good, just some cleanup needed. The main things are in app.tsx but the comments in the rest might help

Vaperizer2 commented:
Also check that all the designs look fine on mobile ( add --host to the run dev script in the package.json to open it on your phone or use the dev tools to adjust screen size)

VissitingLiam committed:
Updated SVG

Added new updated SVG
made Hero Component
made Made CPUanimation tsx slong with Logo as well

VissitingLiam committed:
Css Hero Glos

Updated Hero Css to work in tandem with Hero tsx.
Added a glos for lines goin over the words

VissitingLiam committed:
Cable SVG update

Fixed one of the cables

VissitingLiam committed:
Animation Adjustments

Made Global variables, time and repeated amount

Changed the gloss so that it only does the inline of the text

added a missing cable

IncredibleZuess commented on src/components/custom/SyncCraftCpuAnimation.tsx:
Could we maybe isolate this to an svg file rather than having a massive tsx file. ts will automatically minify this file as it is now but svgs have a separate minifier that makes them very small. 

I know the entire reason for this file is that we can control and reuse these animations all over the website but in this case it is just simpler to have a defacto standard svg.

The great thing about svgs is that you can queue animations as such
```
 <svg width="100%" height="100" xmlns="http://www.w3.org/2000/svg">
  <circle cx="50" cy="50" r="50" style="fill:red;">
    <animate
      attributeName="cx"
      begin="0s"
      dur="8s"
      from="50"
      to="90%"
      fill="freeze" />
  </circle>
</svg> 
```

IncredibleZuess reviewed (commented):
These svgs look good we just need them all to be svgs and not a mix of components and svgs.

# Note on SVGs
SVGs in their nature are just big due to how they work. Expect to see a lot of lines being added when a svg is crafted by hand.
